packages/shared/dropdown/singleselectdropdown.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Select, SelectVariant } from '@patternfly/react-core';
+
+export const SingleSelectDropdown: React.FC<SingleSelectDropdownProps> = ({
+  onChange,
+  selectOptions,
+  selectedKey = '',
+  valueLabelMap,
+  ...props
+}) => {
+  const { t } = useTranslation('plugin__odf-console');
+
+  const [isOpen, setOpen] = React.useState(false);
+  const onSelect = (event: React.MouseEvent | React.ChangeEvent, selection: string) => {
+    /**
+     * For case when the dropdownitem that we want to show on UI (label)
+     * and its corresponding value that we want to store in the redux-state is different. 
+     * e.g: OSDSizeDropdown
+     */
+    const value = valueLabelMap
+                  ? Object.keys(valueLabelMap).find(key => valueLabelMap[key] === selection)
+                  : selection;
+    onChange(value);
+    setOpen(false);
+  };
+
+  return (
+    <div>
```

Comment:
```suggestion
    <div>
```
Why is this div required?